source/winKernel.py

```diff
@@ -479,3 +479,14 @@ def SetThreadExecutionState(esFlags):
 	if not res:
 		raise WinError()
 	return res
+
+
+def LCIDToLocaleName(windowsLCID: LCID) -> Optional[str]:
+	bufSize = 32

```

Comment:
This hard-coded buffer size is too small which can potentially result in some locale name being truncated. Please note the following remark from the [LCIDToLocaleName documentation](https://docs.microsoft.com/en-us/windows/win32/api/winnls/nf-winnls-lcidtolocalename):
> [in] cchName
Size, in characters, of the locale name buffer. The maximum possible length of a locale name, including a terminating null character, is LOCALE_NAME_MAX_LENGTH. This is the recommended size to supply for this parameter.

Note that [the documentation for LOCALE_NAME_MAX_LENGTH.](https://docs.microsoft.com/en-us/windows/win32/intl/locale-name-constants) says:
> Maximum length of a locale name. The maximum number of characters allowed for this string is 85, including a terminating null character.


Rather than hard-coding the size to 86 I'd prefer to follow the approach used for `languageHandler.englishLanguageNameFromNVDALocale` and friends i.e.:
- Call the function first with size set to 0 and buffer set to `None` and save the required size into a variable
- Allocate the buffer of the necessary size and
- Call the function with the buffer created in step 2 and size retrieved in step 1

Reply:
Thanks for pointing this out, I just moved the code without thinking enough.

This is fixed in f1e3451